src/Controller/Entry/EntryFrontController.php

```diff
@@ -5,213 +5,129 @@
-                    true
-                ) ? Criteria::AP_LOCAL : Criteria::AP_ALL
-            )
-            ->setTime($criteria->resolveTime($time))
-            ->setType($criteria->resolveType($type));
-
-        if (null !== $user && 0 < \count($user->preferredLanguages)) {
-            $criteria->languages = $user->preferredLanguages;
+        if ('threads' === $content) {
+            $criteria = new EntryPageView($this->getPageNb($request));
+            $criteria->setContent('threads');
+        } elseif ('microblog' === $content) {
+            $criteria = new PostPageView($this->getPageNb($request));
+            $criteria->setContent('microblog');
+        } else {
+            throw new \LogicException('Invalid content '.$content);
         }
 
-        $method = $criteria->resolveSort($sortBy);
-        $posts = $this->$method($criteria);
-
-        $posts = $this->handleCrossposts($posts);
-
-        if ($request->isXmlHttpRequest()) {
-            return new JsonResponse(
-                [
-                    'html' => $this->renderView(
-                        'entry/_list.html.twig',
-                        [
-                            'entries' => $posts,
-                        ]
-                    ),
-                ]
-            );
-        }
-
-        return $this->render(
-            'entry/front.html.twig',
-            [
-                'entries' => $posts,
-            ]
-        );
-    }
-
-    #[IsGranted('ROLE_USER')]
-    public function subscribed(?string $sortBy, ?string $time, ?string $type, Request $request): Response
-    {
-        $user = $this->getUserOrThrow();
-
-        $criteria = new EntryPageView($this->getPageNb($request));
         $criteria->showSortOption($criteria->resolveSort($sortBy))
-            ->setFederation(
-                'false' === $request->cookies->get(
-                    ThemeSettingsController::KBIN_FEDERATION_ENABLED,
-                    true
-                ) ? Criteria::AP_LOCAL : Criteria::AP_ALL
-            )
+            ->setFederation($federation)
             ->setTime($criteria->resolveTime($time))
             ->setType($criteria->resolveType($type));
-        $criteria->subscribed = true;
 
-        if (0 < \count($user->preferredLanguages)) {
-            $criteria->languages = $user->preferredLanguages;
+        if ('sub' === $subscription) {
+            $this->denyAccessUnlessGranted('ROLE_USER');
+            $user = $this->getUserOrThrow();
+            $criteria->subscribed = true;
+        } elseif ('mod' === $subscription) {
+            $this->denyAccessUnlessGranted('ROLE_USER');
+            $criteria->moderated = true;
+        } elseif ('fav' === $subscription) {
+            $this->denyAccessUnlessGranted('ROLE_USER');
+            $criteria->favourite = true;
+        } elseif ($subscription && 'all' !== $subscription) {
+            throw new \LogicException('Invalid subscription filter '.$subscription);
         }
```

Comment:
So... this if statement is a consequence of Criteria handling the subscription filters with a few boolean variables instead of a single choice-among-options variable like all the other filters are. Honestly in my opinion it'd be better to clean that up into a single choice-among-options variable instead, but like I was saying for some other things I had to sort of draw the line in terms of how much other refactoring to wander around doing as part of the UI change. My thinking was more or less, as long as I've refactored some stuff to improve it overall, I'm not obligated to go around refactoring every bit of code I'm touching (e.g. in this case condensing ->subscribed and ->moderated and ->favourite all down into one variable). With the three separate variables, I'm not real sure how I could eliminate the branching if here with some duplication among the different pathways. The further-ahead solution I think would be to condense all those variables into one at which point the duplication would go away. What do you think?

(As a separate issue, I realized looking at this that we need to be setting $user on the other pathways also; I'll check in a change to do that.)

Reply:
No you are not obligated to refactor everything you touch. This code is just really got to move into a small method so it is not duplicated :sweat_smile: